src/lib/pages/backoffice/Item/ItemForms/ItemEditor/ItemForm/ItemForm.js

```diff
@@ -114,6 +135,8 @@ export class ItemForm extends Component {
         successSubmitMessage={successSubmitMessage}
         title={title}
         pid={pid}
+        modal={hasModal ? modal : null}
```

Comment:
If we go with this, I will then do something different:
```
BaseForm
  onSubmit=() => modal and then call the original on submit if yes
```
basically, I am not sure we want to implement a modal in the base form... only if it can be reused?

Or, if we implement the modal there, then I would specialize it:
```
<BaseForm
    onSubmitCreateModal
    onSubmitEditModal
```
but then the conditions can differ.

Reply:
Yes, it is reusable. Any component that uses the `BaseForm` component will be able to use the modal by passing 2 props:

1. **modal:** It's an obejct that contains the header and the content of the modal, f.e:
```
const modal = {
      header: 'Possible inconsistency',
      content:
        'You are about to change the referred document of this item. This item already has past loans that belongs to the current document. Be aware that changing the document can provoke some inconsistencies. Do you want to proceed ?',
    };
```
2. **shouldModalDisplay:**  This prop allows to pass a function to add some conditions if needed for displaying the modal. If no conditions are needed, just return true